### ISSUE TITLE
Not sure how to apply this...

### ISSUE THREAD

**Issue by dzearing** [brillout/gulp-jspm#9]
I'm trying to apply this plugin in a real scenario.

In order for a non self-executing bundle to be consumed on demand without using a script tag to load it, the client needs a config.js that is updated with the bundle information. That way require statements can resolve modules to the output'd bundle.

This is normally done on the command line via --inject command.

How do i get this process working with this plugin? So that i can have an updated config that the html page can consume to know what's actually in the bundle?

**Comment by brillout** [comment]
will
```
gulp.src('src/main.js')
    .pipe(gulp_jspm({inject: true})) // `jspm bundle main --inject`
    .pipe(gulp.dest('build/'));
```
work out for your use case?

**Comment by brillout** [comment]
feel free to re-open

**Comment by telekosmos** [comment]
For me, inject is not working either. Just in the demo example, if you create a task like exampled above, you get a big exception because of the use of temporary files (I filed https://github.com/jspm/jspm-cli/issues/1183 and you can get a small explanation on it). 

I was trying to upgrade and things got tricky. When you run `jspm.bundle(...)` with `inject: true`, the jspm API updates the __config.js__ file to add the `bundles` property. This is just how `jspm` works when you run `jspm bundle main --inject` in __demo__ dir, you get a __build.js__ and __build.js.map__ in __src__ dir (=`baseURL` from __package.json__) and the modified __config.js__ will look like:
```javascript
System.config({
  defaultJSExtensions: true,
  transpiler: "traceur",
  paths: {
    "github:*": "jspm_packages/github/*",
    "npm:*": "jspm_packages/npm/*"
  },
  bundles: {
    "main.bundle.js": [
      "main.js",
      "message.js"
    ]
  },
// ... more mappings here
});
```
Keeping the jspm API and using normal files instead the temporary ones fix the first issue, but then you have to deal with `gulp.dest(<path>)` as the bundle file it has to be '_in sync_' with the `bundles` property in __config.js__ file.

So far, and I can be (very) wrong, using the plugin without appending a `gulp.dest(...)` seems to work, but you just can yield the 'default' bundle. Adding parameters to the plugin (via i.e. [gulp-param](https://github.com/stoeffel/gulp-param)) can solve this issue keeping the original concept.

Another chance, related what is commented in #14, could be using [systemjs builder](https://github.com/systemjs/builder), but AFAIK the _config.js_ issue with `bundles` property is not solved on inject :-/.

Sorry for the long speech, critics and clarification are welcome ;-)

**Comment by dzearing** [comment]
Yeah inject didn't work due to the temp files. I tried modifying the plugin, but ended up not going this route. Instead I just use systemjs builder APIs directly and build out a gulp bundle task that takes in a bundle.json config, and for each bundle description, creates a bundle and injects bundle metadata into a given config.

**Comment by brillout** [comment]
@telekosmos @dzearing I'll have a look at it sometime, hopefully this week. Thanks for posting.

**Comment by brillout** [comment]
once #16 finished, we would have

 - in memory building
 - have an option to `gulp-jspm` that specifies the relative path of the bundle

which should solve this problem.

**Comment by Sri276286** [comment]
This is what I am using 

gulp.task('jspm', function() {
    jspm({
            // configOverride: {
            //     baseURL: 'target/webapp'
            // },
            bundles: [{
                src: ['target/webapp/main.js'],
                dst: 'app-main.js'
            }]

        }).on('error', function(e) {
            console.log(e);
        })
        .pipe(gulp.dest('dist-bundle'));
});

--------------

I am getting an error : 

TypeError: Cannot read property 'bundles' of null 

Please let me know how to fix this.
